pkg/catalog/loader/loader.go

```diff
@@ -465,6 +470,22 @@ func (store *Store) LoadTemplatesWithTags(templatesList, tags []string) []*templ
 						stats.Increment(templates.SkippedUnsignedStats)
 						return
 					}
+
+					if parsed.SelfContained &&
+						store.config.ExecutorOptions.Options.Vars.IsEmpty() && !store.config.ExecutorOptions.Options.EnvironmentVariables &&
+						templateContainsUnresolvedVariables(templatePath) {
+						stats.Increment(templates.SkippedSelfContainedStats)
+						return
+					}
+
+					if parsed.HasFileProtocol() &&
+						lo.NoneBy(store.config.ExecutorOptions.Options.Targets, func(target string) bool {
+							return fileutil.FileOrFolderExists(target)
+						}) {
+						stats.Increment(templates.SkippedFileStats)
+						return
+					}
+
```

Comment:
not sure if we should compute this check here , the stats definetly belong here but checking everytime if target is a file for every template might increase latency on small hosts/boxes . i think it would be better if we precalcuate and put it in types.Options ( we do this for headless,DAST and other filtering ) so we could also add 2 options
```
HasGlobalVars bool
HasFileInput bool
```
> or something similar

another reason for this is that nuclei supports target from various flags `-u , -l` and even input mode and this logic currently only checks for values of `-u` and not `-l`